### PR TITLE
[proj4] Fix the feature database judgment

### PR DESCRIPTION
**Pull request by JackBoosY**
Related: #8757.

### CODE REVIEW

File: ports/proj4/portfile.cmake [hunk 1]
```diff
@@ -23,20 +21,21 @@ endif()
...
         set(BIN_SUFFIX .exe)
-        if (VCPKG_TARGET_ARCHITECTURE STREQUAL arm)
-            if (NOT EXISTS ${CURRENT_INSTALLED_DIR}/../x86-windows/tools/sqlite3-bin.exe)
-                message(FATAL_ERROR "Proj4 database need to install sqlite3[tool]:x86-windows first.")
-            endif()
-            set(SQLITE3_BIN_PATH ${CURRENT_INSTALLED_DIR}/../x86-windows/tools)
-        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL arm64 OR (VCPKG_TARGET_ARCHITECTURE STREQUAL x64 AND VCPKG_LIBRARY_LINKAGE STREQUAL dynamic))
-            if (NOT EXISTS ${CURRENT_INSTALLED_DIR}/../x64-windows/tools/sqlite3-bin.exe)
-                message(FATAL_ERROR "Proj4 database need to install sqlite3[tool]:x64-windows first.")
+        if (VCPKG_TARGET_ARCHITECTURE MATCHES arm OR VCPKG_TARGET_ARCHITECTURE MATCHES uwp)
```
Comment (ras0219-msft) [inline ports/proj4/portfile.cmake]: `sqlite3-bin.exe` is completely independent of the current target architecture.

Instead, this entire block (lines 27-41) should test:
1. If `${CURRENT_INSTALLED_DIR}/../x86-windows/tools/sqlite3-bin.exe` exists, use that.
2. If `${CURRENT_INSTALLED_DIR}/../x86-windows-static/tools/sqlite3-bin.exe` exists, use that.
3. If `${CURRENT_INSTALLED_DIR}/../x64-windows/tools/sqlite3-bin.exe` exists and `CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64"`, use it.
4. If `${CURRENT_INSTALLED_DIR}/../x64-windows-static/tools/sqlite3-bin.exe` exists and `CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64"`, use it.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/proj4/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/proj4/portfile.cmake]: We have established a dependency with `sqlite3 [tool]`, so if the current triplet is not part of arm or uwp, we can use _VCPKG_PATH/installed/CURR_TRIPLET/tools/ sqlite3-bin.exe_ directly.

Also, if the tool is not built on `x86-windows / x86-windows-static / x64-windows / x64-windows-static`, it needs to be processed.

**Review by PhoebeHui — APPROVED** [review]

**Comment by ras0219-msft** [comment]
The `ppconsul` failure looks unrelated; LGTM.

**Pull request merged**